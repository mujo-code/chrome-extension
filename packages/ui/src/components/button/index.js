import { omitKeys } from '@mujo/box'
import { Tracker } from '@mujo/utils'
import React, { useState } from 'react'
import { Box } from '../box'
import { headerS } from '../fonts/styles'
import { ToolTip } from '../tool-tip'

const { track } = Tracker

const colors = {
  primary: {
    color: 'mischka',
    backgroundColor: 'outerSpace',
    highlight: 'saltBox',
  },
  secondary: {
    color: 'white',
    backgroundColor: 'saltBox',
    highlight: 'gravel',
  },
  tertiary: {
    color: 'outerSpace',
    backgroundColor: 'mischka',
    highlight: 'saltBox',
  },
}

export const Button = props => {
  const { children = 'Label', design = 'primary', alt, altOffset = 0 } = props
  const restProps = omitKeys(
    props,
    'design',
    'alt',
    'children',
    'altOffset',
    'onClick'
  )
  const [tooltipOpen, setToolTipOpen] = useState(false)
  const { color, backgroundColor, highlight } = colors[design]
  return (
    <Box
      font="headerS"
      outlineColor={highlight}
      color={color}
      backgroundColor={backgroundColor}
      display="inlineFlex"
      flexDirection="column"
      alignItems="center"
      paddingTop="xs"
      paddingLeft="l"
      paddingBottom="xs"
      paddingRight="l"
      border="none"
      borderRadius="l"
      textDecoration="none"
      position="relative"
      onClick={(e, ...args) => {
        props.onClick && props.onClick(e, ...args)
        track({
          category: 'button',
          action: 'click',
          label: typeof children === 'string' ? children : e.target.textContent,
        })
      }}
      onMouseLeave={() => setToolTipOpen(false)}
      onMouseEnter={() => setToolTipOpen(true)}
      {...restProps}
    >
      {children}
      {alt ? (
        <ToolTip isOpen={tooltipOpen} offset={altOffset}>
          {alt}
        </ToolTip>
      ) : null}
    </Box>
  )
}

Button.defaultProps = { Component: 'button' }
