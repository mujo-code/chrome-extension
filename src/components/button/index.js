import { Box } from '@jcblw/box'
import { removeKeys } from '@jcblw/box/dist/lib/remove-keys'
import React, { useState } from 'react'
import { headerS } from '../fonts/styles'
import { ToolTip } from '../tool-tip'

const colors = {
  primary: {
    color: 'mischka',
    backgroundColor: 'outerSpace',
    highlight: 'saltBox',
  },
  secondary: {
    color: 'white',
    backgroundColor: 'saltBox',
    highlight: 'gravel',
  },
  tertiary: {
    color: 'outerSpace',
    backgroundColor: 'mischka',
    highlight: 'saltBox',
  },
}

export const Button = props => {
  const {
    children = 'Label',
    design = 'primary',
    alt,
    altOffset = 0,
  } = props
  const restProps = removeKeys(
    props,
    'design',
    'alt',
    'children',
    'altOffset'
  )
  const [tooltipOpen, setToolTipOpen] = useState(false)
  const { color, backgroundColor, highlight } = colors[design]
  return (
    <Box
      outlineColor={highlight}
      color={color}
      backgroundColor={backgroundColor}
      display="inlineFlex"
      direction="column"
      alignItems="center"
      paddingTop="xs"
      paddingLeft="l"
      paddingBottom="xs"
      paddingRight="l"
      border="none"
      borderRadius="l"
      textDecoration="none"
      position="relative"
      onMouseLeave={() => setToolTipOpen(false)}
      onMouseEnter={() => setToolTipOpen(true)}
      {...headerS}
      {...restProps}
    >
      {children}
      {alt ? (
        <ToolTip isOpen={tooltipOpen} offset={altOffset}>
          {alt}
        </ToolTip>
      ) : null}
    </Box>
  )
}

Button.defaultProps = { Component: 'button' }
