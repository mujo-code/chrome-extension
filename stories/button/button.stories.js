import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import React from 'react'
import { Button } from '../../src/components/button'

storiesOf('Button', module).add('with text', () => (
  <Button onClick={action('clicked')}>Hello Mujo</Button>
))
