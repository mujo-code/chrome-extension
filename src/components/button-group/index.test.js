import React from 'react'
import { create } from 'react-test-renderer'
import { Button } from '../button'
import { ButtonGroup } from '.'

test('ButtonGroup component matches snapshot', () => {
  const tree = create(
    <ButtonGroup>
      <Button>Foo</Button>
      <Button>Bar</Button>
    </ButtonGroup>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
