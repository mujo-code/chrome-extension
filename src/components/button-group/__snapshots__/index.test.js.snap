// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup component matches snapshot 1`] = `
<div
  className="css-1xhj18k"
>
  <button
    backgroundColor="outerSpace"
    borderBottomLeftRadius="l"
    borderBottomRightRadius="none"
    borderTopLeftRadius="l"
    borderTopRightRadius="none"
    className="css-9nurzw"
    color="mischka"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    outlineColor="saltBox"
    position="relative"
  >
    Foo
  </button>
  <div
    className="css-1yljtvz"
  />
  <button
    backgroundColor="outerSpace"
    borderBottomLeftRadius="none"
    borderBottomRightRadius="l"
    borderTopLeftRadius="none"
    borderTopRightRadius="l"
    className="css-9nurzw"
    color="mischka"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    outlineColor="saltBox"
    position="relative"
  >
    Bar
  </button>
</div>
`;
